src/Transformer/DecimalTransformer.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the API Extension project.
+ *
+ * (c) Vincent Chalamon <vincentchalamon@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiExtension\Transformer;
+
+use Doctrine\DBAL\Types\Type;
+
+/**
+ * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ */
+final class DecimalTransformer implements TransformerInterface
+{
+    public function supports(array $mapping, $value): bool
+    {
+        return in_array($mapping['type'], [Type::DECIMAL], true);
+    }
+
+    public function toObject(array $mapping, $value): float
+    {
+        return (float) $value;
```

Comment:
Sadly, Doctrine parse it as `string`, so we should keep it like that

Reply:
`toScalar` is used in `ApiContext` so there is no need for change here :)